File: frontend/src/services/supportService.js
import { apiClient } from './apiClient';

export const supportService = {
  createTicket: async (ticketData) => {
    return await apiClient.post('/support/tickets', ticketData);
  },

  getUserTickets: async () => {
    return await apiClient.get('/support/tickets');
  },

  getTicket: async (ticketId) => {
    return await apiClient.get(`/support/tickets/${ticketId}`);
  },

  addMessage: async (ticketId, messageData) => {
    return await apiClient.post(`/support/tickets/${ticketId}/messages`, messageData);
  },

  getTicketMessages: async (ticketId) => {
    return await apiClient.get(`/support/tickets/${ticketId}/messages`);
  }
};

File: Backend/src/main/java/com/tripnest/support/service/SupportService.java
package com.tripnest.support.service;

import com.tripnest.support.dto.SupportTicketRequest;
import com.tripnest.support.dto.SupportTicketResponse;
import com.tripnest.support.dto.TicketMessageRequest;
import com.tripnest.support.dto.TicketMessageResponse;

import java.util.List;

public interface SupportService {
    SupportTicketResponse createTicket(String username, SupportTicketRequest request);
    List<SupportTicketResponse> getUserTickets(String username);
    SupportTicketResponse getTicket(String username, Long ticketId);
    TicketMessageResponse addMessage(String username, Long ticketId, TicketMessageRequest request);
    List<TicketMessageResponse> getTicketMessages(String username, Long ticketId);
}
